client/java/src/main/java/com/linecorp/centraldogma/internal/client/ReplicationLagTolerantCentralDogma.java

```diff
@@ -681,6 +683,9 @@ private boolean handleRevisionNotFound(
             String projectName, String repositoryName, Revision revision, Throwable cause) {
 
         requireNonNull(cause, "cause");
+        if (cause instanceof CompletionException) {
+            cause = Throwables.getRootCause(cause);
```

Comment:
root cause might be irrelevant, e.g. if someone throws a `RevisionNotFoundException(cause=IOException)`